source/MQTTFileDownloader.c

```diff
@@ -81,14 +97,38 @@ static MQTTFileDownloaderStatus_t handleCborMessage( uint8_t * decodedData,
  * @param[in] messageLength Length of the MQTT message received.
  * @param[out] decodedData Buffer to place the decoded data in.
  * @param[out] decodedDataLength Length of decoded data.
+ * @param[out] fileId ID of the file to which the data block belongs.
+ * @param[out] blockId ID of the received block.
+ * @param[out] blockSize Size of the receive block in bytes.
  *
  * @return uint8_t returns appropriate MQTT File Downloader Status.
  */
-static MQTTFileDownloaderStatus_t handleJsonMessage( uint8_t * decodedData,
+static MQTTFileDownloaderStatus_t handleJsonMessage( int32_t * fileId,
+                                                     int32_t * blockId,
+                                                     int32_t * blockSize,
+                                                     uint8_t * decodedData,
                                                      size_t * decodedDataLength,
                                                      uint8_t * message,
                                                      size_t messageLength );
 
+/**
+ * @brief Handles and decodes the received message in JSON format.
+ *
+ * @param[in] str String which has the ASCII representation of the
+ *            number.
+ * @param[in] len Length of the string.
+ * @param[out] num Pointer to an int32_t variable which will be filled
+ *            with the parsed value. If the return value of the function
+ *            call is -1, then the value present in num is invalid.
+ *
+ * @return int32_t if the string is malformed or the decoded number
+ * cannot fit in an int32_t value then this function returns -1. On
+ * successful parsing, it returns 0.
+ */
+static int32_t getNumberFromString( char * str,
```

Comment:
If this can only ever return `-1` or `0` then why not just have it return a `bool`?

Reply:
Fixed with https://github.com/aws/aws-iot-core-mqtt-file-streams-embedded-c/pull/23/commits/f31c1fd9d4701eb1d079f59f7c8436425172b626